### ISSUE TITLE
You might find my bachelor's thesis interesting

### ISSUE THREAD

**Issue by taktoa** [egraphs-good/egg#46]
Here's a link: http://taktoa.me/eqsat/Thesis.pdf

Unfortunately, my implementation didn't get very far. UIUC's senior thesis credit grades based on what you write, not what you implement.

**Comment by mwillsey** [comment]
Looks awesome at a first glance! It's on my todo list. I'll get back to you later with more details.

**Comment by mwillsey** [comment]
Hey Remy! I have not forgotten about this! I've only been able to take a skim so far, but it looks really cool.

I'm going to close this issue for now, but I've just opened the [discussions](https://github.com/egraphs-good/egg/discussions) feature on this repo if you'd like to start a thread on your work (even if just to show it off!). In particular, I'd love to hear if you think it would be worthwhile to bring any of the innovations from your work into egg
